### PR TITLE
[ALLUXIO-2119] Remove use of Whitebox in UnderFileSystemFileInStream Test

### PR DESCRIPTION
**Pull request by aaudiber**
https://alluxio.atlassian.net/browse/ALLUXIO-2119

### CODE REVIEW

**Comment by aaudiber** [comment]
alluxio-bot, check this please

File: core/client/src/test/java/alluxio/client/file/UnderFileSystemFileInStreamTest.java [hunk 1]
```diff
@@ -139,7 +144,5 @@ public void readToEOFTest() throws Exception {
       Assert.assertEquals((byte) i, b[i]);
     }
     Assert.assertEquals(-1, mTestStream.read());
```
Comment (calvinjia) [inline core/client/src/test/java/alluxio/client/file/UnderFileSystemFileInStreamTest.java]: Is there any way to check the eof flag?

File: core/common/src/test/java/alluxio/ConfigurationRule.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ */
+public class ConfigurationRule implements TestRule {
+  private final Map<String, String> mKeyValuePairs;
+
+  /**
+   * @param keyValuePairs map from configuration keys to the values to set them to
+   */
+  public ConfigurationRule(Map<String, String> keyValuePairs) {
+    mKeyValuePairs = keyValuePairs;
+  }
```
Comment (calvinjia) [inline core/common/src/test/java/alluxio/ConfigurationRule.java]: Is it guaranteed there will only be one configuration rule executed at a time (without interleaving)?

Comment (aaudiber) [inline core/common/src/test/java/alluxio/ConfigurationRule.java]: As long as people use them like `@Rule`s there will be no interleaving

File: core/client/src/test/java/alluxio/client/file/UnderFileSystemFileInStreamTest.java (hunk 1 above)
Comment (aaudiber) [inline core/client/src/test/java/alluxio/client/file/UnderFileSystemFileInStreamTest.java]: Right, the -1 is the external functionality and the `mEOF` is internal. We should only test external.

**Comment by calvinjia** [comment]
LGTM

**Pull request merged**